app/javascript/packs/application/components/CreateTourForm.js

```diff
@@ -41,46 +42,7 @@ export class CreateTourForm extends Component {
   render() {
     return (
       <form onSubmit={this.handleSubmit}>
-        <label htmlFor="title">Title: </label>
-        <input
-          name="title"
-          type="text"
-          value={this.state.title}
-          onChange={this.handleChange}
-        />
-        <br />
-        <label htmlFor="starting_point">Starting Point: </label>
-        <input
-          name="starting_point"
-          type="text"
-          value={this.state.starting_point}
-          onChange={this.handleChange}
-        />
-        <br />
-        <label htmlFor="directions">Directions: </label>
-        <input
-          name="directions"
-          type="text"
-          value={this.state.directions}
-          onChange={this.handleChange}
-        />
-        <br />
-        <label htmlFor="estimated_time">Estimated Time: </label>
-        <input
-          name="estimated_time"
-          type="text"
-          value={this.state.estimated_time}
-          onChange={this.handleChange}
-        />
-        <br />
-        <label htmlFor="description">Description: </label>
-        <input
-          name="description"
-          type="text"
-          value={this.state.description}
-          onChange={this.handleChange}
-        />
-        <br />
+        <TourFormFields onChange={this.handleChange} tour={this.state} />
```

Comment:
Nice!

Reply:
I actually think that the `TourFormFields` should handle the `change` event and manage the state. The only difference between the `Create` and `Edit` will likely be the submission.

However, the `CreateTourForm` should handle the submit. So, I actually think this line should look like this:

```javascript
<TourFormFields onSubmit={this.handleSubmit} initialValues={this.props.tour} />
```

Also, note the `initialValues`. Best practices dictate that a prop that is used to initialize state should be prefixed with `initial`